torch/_inductor/codegen/common.py

```diff
@@ -285,10 +286,18 @@ def _print_Pow(self, expr):
         # in sizevar compute.  Instead of adding support for floating
         # point pow, you should make upstream retranslate the Sympy expression
         # into Tensor expressions earlier and do that instead.
-        if exp == 0.5:
-            return self._helper_sqrt(base)  # type: ignore[attr-defined]
-        elif exp == -0.5:
-            return "1/" + self._helper_sqrt(base)  # type: ignore[attr-defined]
+
+        if abs(exp) < 1.0 and exp != 0:
+            pow2 = math.log(1 / abs(exp), 2)
+            # call nested sqrt
+            if pow2.is_integer():
+                for _ in range(int(pow2)):
+                    base = self._helper_sqrt(base)  # type: ignore[attr-defined]
```

Comment:
At this point would it not make more sense to just codegen a call to `pow`. Replacing pow with a loop of sqrt calls doesn't really make sense performance-wise.

Reply:
In the case I was fixing, it was only handling the additional -0.25 case i.e. 1/sqrt(sqrt(x)). Maybe we can just add a max nested depth? I'm not too clear why codegen pow is such a bad idea for Triton. But cpp has the same restriction.

Reply:
Maybe @jansel can comment but I believe there was a principled objection to having floating point in indexing calculations generally.  There is no reason we couldn't have a `_helper_pow` that works the same as `_helper_sqrt`.

Reply:
These are the times I measured in CUDA:

```
sqrt_2_pow_op in 0.366592 ms.
sqrt_3_pow_op in 0.206848 ms.
reciprocal_sqrt_2_pow_op in 0.185344 ms.
reciprocal_sqrt_3_op in 0.18432 ms.

sqrt_2_op in 0.319488 ms.
sqrt_3_op in 0.423936 ms.
reciprocal_sqrt_2_op in 0.431104 ms.
reciprocal_sqrt_3_op in 0.531456 ms.

```

Looks like powf is faster when the exponent is negative smaller and or negative

So yes, using pow is probably better.